src/deepsparse/v2/middleware/timer_middleware.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+import time
+from functools import wraps
+from typing import Optional
+
+from deepsparse.v2.utils.state import InferenceState
+
+
+class Timer:
+
+    _lock = threading.Lock()
```

Comment:
lock should be local to a single timer object, not static across all
why should one timer be able to block another when writing to separate timing dictionaries?